pkg/runner/flags_test.go

```diff
@@ -242,10 +241,97 @@ func TestParseFlagsConfigPrecedence(t *testing.T) {
 			result: func(c Config) Config { return c },
 		},
 		{
-			name:               "Config file only",
-			configFileContents: "web.listen-address: localhost:9201",
+			name: "Config file only",
```

Comment:
Consider adding a test case for using both `startup.dataset` and `startup.dataset.config` at the same time to make sure we handle it correctly.